### PR TITLE
Replace go.uber.org/atomic with sync/atomic

### PR DESCRIPTION
**Pull request by serprex**
## What changed?
go 1.19 introduced atomic types
https://tip.golang.org/doc/go1.19#atomic_types

## Why?
Reduces third party dependencies by consolidating on stdlib

## How did you test it?
Could build with `make bins` & `make functional-test`,
but can't run tests on my system since I don't have docker

## Potential risks
The hairiest part is `atomic.Pointer` but that's only used in tests

## Is hotfix candidate?
No

### CODE REVIEW

File: common/persistence/client/fault_injection_errorgenerator.go
Comment (rodrigozhou) [inline common/persistence/client/fault_injection_errorgenerator.go:147]: You should load the value once before using it. Otherwise, there's a risk of the rate changing and the message in L147 showing a value not corresponding to the comparison in L141. Issue not introduced in this PR though.

File: common/quotas/dynamic.go
Comment (rodrigozhou) [inline common/quotas/dynamic.go:127]: Do you even need to init them? Why not this?
```suggestion
	d := &MutableRateBurstImpl{}
```

File: common/persistence/client/fault_injection_errorgenerator.go
Comment (rodrigozhou) [inline common/persistence/client/fault_injection_errorgenerator.go:112]: Is it better to convert to int vs having `atomic.Pointer[float64]`?
Also, is the std `atomic` module any better than the `uber` one? Or is it just to reduce deps?

File: common/quotas/dynamic.go
Comment (serprex) [reply common/quotas/dynamic.go:127]: That works

File: common/persistence/client/fault_injection_errorgenerator.go
Comment (serprex) [reply common/persistence/client/fault_injection_errorgenerator.go:112]: Just reduce deps mostly, uber's atomic wraps `sync/atomic`

File: common/persistence/client/fault_injection_errorgenerator.go
Comment (serprex) [reply common/persistence/client/fault_injection_errorgenerator.go]: Fixed issue

**Review by pdoerner — APPROVED** [review]
Sorry for late review. LGTM. Approving, but please double check no new usages of go.uber.org/atomic have been added recently before merging.

**Comment by serprex** [comment]
I looked over latest merge, seems fine

**Comment by serprex** [comment]
@pdoerner I can't merge this, someone on the temporal team would need to merge

**Comment by stephanos** [comment]
@serprex seems like there is a lint error

**Comment by serprex** [comment]
Being flagged for issues in main

**Comment by bergundy** [comment]
Looks like this PR was marked as stale. @serprex I am now subscribed to this PR and would like to merge this if you're willing to do another pass to fix the merge conflicts.

Thanks for contributing!

**Comment by serprex** [comment]
@bergundy done

**Review by bergundy — APPROVED** [review]

**Comment by serprex** [comment]
@bergundy if it's easier to copy branch & create a fresh PR that doesn't have permission issues in CI, feel free to do so

**Comment by bergundy** [comment]
I might try to fix the branch, I saw the error but didn’t have time to dive into the reason yet. I will do my best to merge it this week.

**Pull request closed without merging**
